### PR TITLE
SOLR-15918 Avoid repetitive creating parent znodes

### PR DESCRIPTION
**Pull request by madrob**
https://issues.apache.org/jira/browse/SOLR-15918

When uploading a configset, we do not need to create the parent znodes for each file.

I did not have idea on how to make a unit test here that would fail so that we could verify this, the best I could do was manually inspect the log output.

### CODE REVIEW

File: solr/solrj/src/java/org/apache/solr/common/cloud/SolrZkClient.java [hunk 1]
```diff
@@ -543,6 +543,9 @@ public void makePath(String path, byte[] data, CreateMode createMode,
           throw e;
         }
       } catch (NodeExistsException e) {
+        if (log.isDebugEnabled()) {
```
Comment (anshumg) [inline solr/solrj/src/java/org/apache/solr/common/cloud/SolrZkClient.java:546]: You can skip this check here, as this is parameterized and doing any string ops

Comment (madrob) [reply solr/solrj/src/java/org/apache/solr/common/cloud/SolrZkClient.java:546]: It's calling getPath which makes precommit upset, IIRC

Comment (anshumg) [reply solr/solrj/src/java/org/apache/solr/common/cloud/SolrZkClient.java:546]: that's interesting.

**Comment by madrob** [comment]
@janhoy I would like to include this in 9.0 I feel it is safe to commit.

**Comment by janhoy** [comment]
What would be the benefit of getting this in 9.0? Have you measured a performance benefit?

mid there is no clear benefit or bug fix perhaps this can wait until 9.1?

**Comment by madrob** [comment]
I don't have anything measured, the benefit is reduced reqs to zookeeper. 9.1 is fine

**Pull request merged**